webviz_subsurface/plugins/_tornado_plotter_fmu/shared_settings/_filters.py

```diff
@@ -0,0 +1,156 @@
+import json
+from typing import List, Optional
+
+import webviz_core_components as wcc
+from dash import ALL, Input, Output, callback, callback_context
+from dash.development.base_component import Component
+from webviz_config.webviz_plugin_subclasses import LayoutUniqueId, SettingsGroupABC
+
+from webviz_subsurface._providers import EnsembleTableProvider
+
+from .._plugin_ids import PlugInIDs
+
+
+class Filters(SettingsGroupABC):
+    """Settingsgroup for filters"""
+
+    class IDs:
+        # pylint: disable=too-few-public-methods
+        SINGLE_FILTER = "single-filter"
+        MULTI_FILTER = "multi-filter"
+
+    def __init__(
+        self,
+        table_provider: EnsembleTableProvider,
+        single_filters: List[str],
+        multi_filters: List[str],
+        ensemble_name: str,
+    ) -> None:
+        super().__init__("Filters")
+        self._table_provider = table_provider
+        self._single_filters = single_filters
+        self._multi_filters = multi_filters
+        self._ensemble_name = ensemble_name
+        self._plugin_unique_id = LayoutUniqueId(plugin_uuid=PlugInIDs.PlugIn.PLUGIN_ID)
+        self.single_filter_id = self.register_component_unique_id(
+            Filters.IDs.SINGLE_FILTER
+        )
+        self.multi_filter_id = self.register_component_unique_id(
+            Filters.IDs.MULTI_FILTER
+        )
+
+    def layout(self) -> List[Component]:
+        elements = []
+
+        # Creates dropdowns for any data columns added as single value filters
+        for selector in self._single_filters:
+            values = self._table_provider.get_column_data([selector])[selector].unique()
+
+            elements.append(
+                wcc.Dropdown(
+                    label=selector,
+                    id={
+                        "id": self.single_filter_id,
+                        "name": selector,
+                        "type": "single_filter",
+                    },
+                    options=[{"label": val, "value": val} for val in values],
+                    value=values[0],
+                    clearable=False,
+                )
+            )
+
+        # Creates dropdowns for any data columns added as single value filters
+        for selector in self._multi_filters:
+            values = self._table_provider.get_column_data([selector])[selector].unique()
+            elements.append(
+                wcc.Dropdown(
+                    label=selector,
+                    id={
+                        "id": self.multi_filter_id,
+                        "name": selector,
+                        "type": "multi_filter",
+                    },
+                    options=[{"label": val, "value": val} for val in values],
+                    value=values,
+                    multi=True,
+                )
+            )
+        return elements
+
+    def set_callbacks(self) -> None:
```

Comment:
When wanting to get rid of the stores, these callbacks must either be moved to the plugin class (if you want to continue using shared settings) or to the view (if you rather want to add the settings group to the view). In any case, you'd need to update the components directly without taking the detour via stores (whenever possible).